src/onig-string.cc

```diff
@@ -11,10 +11,15 @@ void OnigString::Init(Local<Object> target) {
 }
 
 NAN_METHOD(OnigString::New) {
-  Nan::HandleScope scope;
-  OnigString* scanner = new OnigString(Local<String>::Cast(info[0]));
-  scanner->Wrap(info.This());
-  info.GetReturnValue().SetUndefined();
+  Local<String> content = Local<String>::Cast(info[0]);
+  if (!content->IsString()) {
+    Nan::ThrowTypeError("Argument must be a string");
+    return;
+  }
+
+  OnigString* string = new OnigString(content);
```

Comment:
Yeah, I think the `ObjectWrap` class causes `delete` to be called in the Javascript object's finalizer.